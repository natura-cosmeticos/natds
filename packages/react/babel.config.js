const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
