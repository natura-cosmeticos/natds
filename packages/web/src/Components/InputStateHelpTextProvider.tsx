import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import { IThemeWeb } from 'Themes';
import { tokens } from '@naturacosmeticos/natds-styles';
import ErrorIcon from '@material-ui/icons/HighlightOffOutlined';
import SuccessIcon from '@material-ui/icons/CheckCircleOutline';

import { getProp, getColorByState, stateStyles } from './TextField/shared';

export interface IInputStateHelpTextProviderProps {
  id?: string;
  theme?: IThemeWeb;
  label?: string;
  helpText?: string;
  required?: boolean;
  disabled?: boolean;
  state: 'error' | 'success' | undefined;
  className?: string;
  children?: any;
  readOnly?: boolean;
}

export const InputStateHelpTextProvider: FunctionComponent<IInputStateHelpTextProviderProps> = (
  props: IInputStateHelpTextProviderProps
) => {
  const {
    id,
    label,
    helpText,
    theme,
    required = false,
    disabled = false,
    state,
    className,
    children,
    readOnly = false,
  } = props;

  const content = label && required ? `${label} *` : label;
  const IconState = stateIcons[String(state)];
  const stateIcon = IconState && <IconState theme={theme} />;

  return (
    <Container theme={theme} className={className}>
      {content && (
        <Label theme={theme} htmlFor={id} state={state} disabled={disabled}>
          {content}
        </Label>
      )}
      {children}
      {!readOnly && helpText && (
        <HelpText theme={theme} state={state} disabled={disabled}>
          {stateIcon}
          {helpText}
        </HelpText>
      )}
    </Container>
  );
};

export default InputStateHelpTextProvider;

const Container = styled.div`
  display: flex;
  flex-flow: column nowrap;
  font-family: ${getProp('typography', 'fontFamily')};
`;

const Label = styled.label`
  font-size: ${getProp('typography', 'subtitle2', 'fontSize')};
  font-weight: ${getProp('typography', 'subtitle2', 'fontWeight')};
  color: ${getColorByState(stateStyles.hover)};
  line-height: 1.2;
  padding: 0 0 ${tokens.spacing.spacingMicro}px;
  font-family: ${getProp('typography', 'subtitle2', 'fontFamily')};
`;

const HelpText = styled.span`
  font-size: ${getProp('typography', 'caption', 'fontSize')};
  font-weight: ${getProp('typography', 'caption', 'fontWeight')};
  color: ${getColorByState(stateStyles.hover)};
  line-height: 1.2;
  padding: ${tokens.spacing.spacingMicro}px 0 0;
  display: flex;
  align-items: center;
  font-family: ${getProp('typography', 'subtitle2', 'fontFamily')};
`;

const baseIcon = `
  width: 16px!important;
  height: 16px!important;
  margin-right: 4px;
`;

const IconError = styled(ErrorIcon)`
  ${baseIcon}
`;

const IconSuccess = styled(SuccessIcon)`
  ${baseIcon}
`;

const stateIcons = {
  error: IconError,
  success: IconSuccess,
};
