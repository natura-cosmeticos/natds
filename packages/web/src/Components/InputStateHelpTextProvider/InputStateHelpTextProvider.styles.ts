import { createStyles } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import { IThemeWeb } from '../../Themes'

const style = ({ sizes }: IThemeWeb) => createStyles({
  icon: {
    height: sizes?.small,
    marginRight: sizes?.micro,
    width: sizes?.small
  }
})

export const useStyles = makeStyles(style, { name: 'NatDSInputStateHelpTextProvider' })

export default useStyles
