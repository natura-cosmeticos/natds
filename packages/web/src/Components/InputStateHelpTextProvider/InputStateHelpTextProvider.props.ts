import * as React from 'react'

export interface IInputStateHelpTextProviderProps {
  children?: React.ReactNode;
  className?: string;
  disabled?: boolean;
  helpText?: string;
  id?: string;
  label?: string;
  readOnly?: boolean;
  required?: boolean;
  state: 'error' | 'success' | undefined;
}
