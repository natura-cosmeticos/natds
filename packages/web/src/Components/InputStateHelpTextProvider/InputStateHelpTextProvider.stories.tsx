import { Meta, Story } from '@storybook/react'
import * as React from 'react'
import { InputStateHelpTextProvider, IInputStateHelpTextProviderProps } from './InputStateHelpTextProvider'
import { FormHelperText } from '../FormHelperText'
import { InputLabel } from '../InputLabel'

export default {
  component: InputStateHelpTextProvider,
  subcomponents: { FormHelperText, InputLabel },
  title: 'Utilities/Input State Help Text Provider'
} as Meta

const Template : Story<IInputStateHelpTextProviderProps> = (args: IInputStateHelpTextProviderProps) => (
  <InputStateHelpTextProvider state={args.state} {...args} />
)

export const Playground : Story<IInputStateHelpTextProviderProps> = Template.bind({})
