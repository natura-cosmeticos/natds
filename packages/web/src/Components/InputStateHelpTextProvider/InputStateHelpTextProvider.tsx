import * as React from 'react'
import { IInputStateHelpTextProviderProps } from './InputStateHelpTextProvider.props'
import { FormHelperText } from '../FormHelperText'
import { InputLabel } from '../InputLabel'
import { FormControl } from '../FormControl'

export { IInputStateHelpTextProviderProps } from './InputStateHelpTextProvider.props'

// eslint-disable-next-line complexity
export const InputStateHelpTextProvider: React.FunctionComponent<IInputStateHelpTextProviderProps> = (props: IInputStateHelpTextProviderProps) => {
  const {
    id,
    label,
    helpText,
    required = false,
    disabled = false,
    state,
    className,
    children,
    readOnly = false
  } = props

  const content = label && required ? `${label} *` : label

  return (
    <FormControl className={className}>
      {content
        && (
        <InputLabel htmlFor={id} error={state === 'error'} disabled={disabled} state={state}>
          {content}
        </InputLabel>
        )}
      {children}
      {!readOnly && helpText
        && (
        <FormHelperText disabled={disabled} error={state === 'error'} state={state}>
          {helpText}
        </FormHelperText>
        )}
    </FormControl>
  )
}

export default InputStateHelpTextProvider
