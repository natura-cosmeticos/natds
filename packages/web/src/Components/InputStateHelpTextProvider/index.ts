import InputStateHelpTextProvider from './InputStateHelpTextProvider'

export { IInputStateHelpTextProviderProps } from './InputStateHelpTextProvider.props'
export { InputStateHelpTextProvider }
export default InputStateHelpTextProvider
