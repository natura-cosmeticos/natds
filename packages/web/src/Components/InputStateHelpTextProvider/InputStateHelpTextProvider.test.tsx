import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import { InputStateHelpTextProvider } from './InputStateHelpTextProvider'

describe('InputStateHelpTextProvider component', () => {
  describe('when state is error', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider state="error" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when state is success', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider state="success" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when readonly property is explicitly true', () => {
    it('and when state is error', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider readOnly state="error" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
    it('and when state is success', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider readOnly state="success" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when readonly property is explicitly false', () => {
    it('and when state is error', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider readOnly={false} state="error" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
    it('and when state is success', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider readOnly={false} state="success" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when help text is set', () => {
    it('and when state is error', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider helpText="Test help text" state="error" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
    it('and when state is success', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <InputStateHelpTextProvider helpText="Test help text" state="success" />
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
})
