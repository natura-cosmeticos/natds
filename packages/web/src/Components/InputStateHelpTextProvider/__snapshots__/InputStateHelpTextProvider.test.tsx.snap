// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputStateHelpTextProvider component when help text is set and when state is error 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
>
  <p
    className="MuiFormHelperText-root NatDSFormHelperText-root-2 NatDSFormHelperText-root-4 Mui-error"
  >
    <i
      aria-hidden={true}
      className="material-icons MuiIcon-root natds-icons natds-icons-outlined-alert-cancel NatDSFormHelperText-icon-3 NatDSIcon-root-5 NatDSIcon-root-6"
    />
    Test help text
  </p>
</div>
`;

exports[`InputStateHelpTextProvider component when help text is set and when state is success 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
>
  <p
    className="MuiFormHelperText-root NatDSFormHelperText-root-2 NatDSFormHelperText-root-7"
  >
    <i
      aria-hidden={true}
      className="material-icons MuiIcon-root natds-icons natds-icons-outlined-alert-check NatDSFormHelperText-icon-3 NatDSIcon-root-5 NatDSIcon-root-8"
    />
    Test help text
  </p>
</div>
`;

exports[`InputStateHelpTextProvider component when readonly property is explicitly false and when state is error 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
/>
`;

exports[`InputStateHelpTextProvider component when readonly property is explicitly false and when state is success 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
/>
`;

exports[`InputStateHelpTextProvider component when readonly property is explicitly true and when state is error 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
/>
`;

exports[`InputStateHelpTextProvider component when readonly property is explicitly true and when state is success 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
/>
`;

exports[`InputStateHelpTextProvider component when state is error should match snapshot 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
/>
`;

exports[`InputStateHelpTextProvider component when state is success should match snapshot 1`] = `
<div
  className="MuiFormControl-root NatDSFormControl-root-1"
/>
`;
